src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -267,29 +272,45 @@ protected Future<Map<String, List<Piece>>> updateOrderAndPoLinesStatus(Map<Strin
     // Once all PO Lines are retrieved from storage check if receipt status
     // requires update and persist in storage
     return getPoLines(poLineIdsForUpdatedPieces, requestContext).compose(poLines -> {
-        // Calculate expected status for each PO Line and update with new one if required
-        // Skip status update if PO line status is Ongoing
-        List<Future<String>> futures = new ArrayList<>();
-        for (PoLine poLine : poLines) {
-          if (poLine.getReceiptStatus() == CANCELLED || poLine.getReceiptStatus() == ONGOING) {
-            continue;
-          }
-          List<Piece> successfullyProcessedPieces = getSuccessfullyProcessedPieces(poLine.getId(), piecesGroupedByPoLine);
-          futures.add(calculatePoLineReceiptStatus(poLine, successfullyProcessedPieces, requestContext)
-            .compose(status -> purchaseOrderLineService.updatePoLineReceiptStatus(poLine, status, requestContext)));
+      // Calculate expected status for each PO Line and update with new one if required
+      // Skip status update if PO line status is Ongoing
+      List<Future<PoLine>> poLinesToUpdate = new ArrayList<>();
+      for (PoLine poLine : poLines) {
+        if (poLine.getReceiptStatus() == CANCELLED || poLine.getReceiptStatus() == ONGOING) {
+          continue;
+        }
+
+        List<Piece> successfullyProcessedPieces = getSuccessfullyProcessedPieces(poLine.getId(), piecesGroupedByPoLine);
+        if (CollectionUtils.isEmpty(successfullyProcessedPieces)) {
+          logger.info("updateRelatedPoLineDetails:: No pieces processed - nothing to update for PO Line '{}'", poLine.getId());
+          continue;
         }
 
-        return collectResultsOnSuccess(futures).map(updatedPoLines -> {
-          logger.debug("{} out of {} PO Line(s) updated with new status", updatedPoLines.size(), piecesGroupedByPoLine.size());
+        Future<PoLine> poLineToUpdateFuture = getPiecesByPoLine(poLine.getId(), requestContext)
+          .map(pieces -> updateRelatedPoLineDetails(poLine, piecesFromStorage.get(poLine.getId()), pieces, successfullyProcessedPieces));
 
-          List<PoLine> successPoLines = StreamEx.of(poLines)
-            .filter(line -> updatedPoLines.contains(line.getId()))
-            .toList();
-          updateOrderStatus.accept(successPoLines);
-          return null;
+        poLinesToUpdate.add(poLineToUpdateFuture);
```

Comment:
at first sight, it is confusing we updated poLine and naming to poLineToUpdate. Maybe need to specify what need to update . poLineToUpdateLocation.

Reply:
> at first sight, it is confusing we updated poLine and naming to poLineToUpdate. Maybe need to specify what need to update . poLineToUpdateLocation.

It updates more than just location though